command/metadata.go

```diff
@@ -0,0 +1,125 @@
+package command
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/mackerelio/mackerel-agent/config"
+	"github.com/mackerelio/mackerel-agent/metadata"
+)
+
+func metadataGenerators(conf *config.Config) []*metadata.Generator {
+	generators := make([]*metadata.Generator, 0, len(conf.MetadataPlugins))
+
+	workdir := os.Getenv("MACKEREL_PLUGIN_WORKDIR")
+	if workdir == "" {
+		workdir = os.TempDir()
```

Comment:
If many metadata plugins are prepared without `MACKEREL_PLUGIN_WORKDIR`, many tempfiles will be created to `os.TempDir`. Is it better to create a subdir and place the tempfiles into it?